From 11e2d89f48d917183eada788378658b6c515f53b Mon Sep 17 00:00:00 2001
From: Carlos Garnacho <carlosg@gnome.org>
Date: Mon, 30 Nov 2015 13:11:40 +0100
Subject: [PATCH] wacom: Fix type of data passed to XIChangeDeviceProperty

For 32bit values we must use long, or Xlib will overread past the
given memory.
---
 plugins/wacom/gsd-wacom-manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/wacom/gsd-wacom-manager.c b/plugins/wacom/gsd-wacom-manager.c
index 2c9b49f..915a095 100644
--- a/plugins/wacom/gsd-wacom-manager.c
+++ b/plugins/wacom/gsd-wacom-manager.c
@@ -532,7 +532,7 @@ reset_touch_buttons (XDevice               *xdev,
 	for (i = 0; buttons[i].button != NULL; i++)
 	{
 		char *propname;
-		int action[2]; /* press + release */
+		long action[2]; /* press + release */
 		Atom prop;
 		int mapped_button = buttons[i].num;
 
-- 
2.5.0

